### DISCUSSION TITLE
xDrip+ nightly update 05/29/2024

### DISCUSSION THREAD

**Discussion by lz2jvc** [NightscoutFoundation/xDrip#3495] [General]
After the last nightly update from today 05/29/2024 of xDrip+, the text of the remaining days of the sensor is now condensed on two lines in the main graph, which is not OK ( picture 2 ). Picture 1 shows how it was before today's update. Now I have two lines of the phone screen occupied by text instead of graphics. Not to mention that with the transitional nightly update from May 22, 2024. the graph has expanded many times horizontally and to return it to its normal position you must select the 6 o'clock display. This is also not OK for me, because the visible field of the graph does not follow the trend of the graph, but you have to manually move it to the right. I guess these things can be fixed, or at least the text, which is already annoying
![2024-05-29 18 42 57](https://github.com/NightscoutFoundation/xDrip/assets/168529053/e0349b00-a4d6-44e7-b44e-55126df64b7a)
![2024-05-29 18 40 50](https://github.com/NightscoutFoundation/xDrip/assets/168529053/7cadfcfa-9490-4831-81b3-8e11ad4d4ed7)
![1688659182437](https://github.com/NightscoutFoundation/xDrip/assets/168529053/e2fa9183-f11c-41a7-ad4f-13c0c8044064)
![1716454557381](https://github.com/NightscoutFoundation/xDrip/assets/168529053/19e10b65-7470-4dce-9c7d-c725d9526042)

**Comment by Navid200** [comment]
Those are two completely different matters.  
  
---  
  
I am not sure why the sensor expiry text now appears in two lines.
@jamorham  
  
---  
  
The time (x) axis range was adjusted recently because of a concern many of us has had with varying slopes shown on the screen.

As diabetics, many of us glance at our graphs frequently during the day.  We do it so often that we don't need to focus every time.  We just see the colors and the slopes and that's it.  A glance may just take less than a second.  And here comes the problem.  If the x and y axis ranges keep changing, it is very easy to miss something wrong with an extremely short glance because the slopes are distorted due to the range change.  

xDrip used to amend the future predictions to the graph extending the range.  That in itself may seem acceptable because you know if you have enabled predictions or not.  But, the problem is that even if you know that you have enabled predictions, you may not always have a low prediction.  If you don't have a low prediction, xDrip has no additional curves to amend.  Then, the x axis shrinks and you have no idea.  It will happen just because of the values in your readings.

I am adding here someone who knows about this.
@trallen 

If we absolutely have to, we can add a new setting.  But, we should not add a setting to take xDrip back to the way it was.  xDrip should not adjust the slopes just because of the readings.
If someone prefers to see the simulated curves that are in the future without having to scroll, I don't see a problem as long as the curve in the future is never going to disappear.
If those curves could go away under some conditions resulting in the x axis changing just because of how your readings change, I strongly advise against it.

I am not a predictive simulation user.
But, I am a user of the forecast low alert.  I knw that the forecast low alert curve in the future disappears if there is no forecasted low.  That is why we cannot have a setting that disables the scroll requirement for forecast low.

What I wonder may work is if we add a new setting that allows us to also specify an offset.  We currently can specify a range.  But, we cannot direct xDrip as to where the reference point should be.
We can have a new setting that lets the user decide to permanently scroll by a few values.  And I absolutely have no idea if @jamorham would approve this.

But, no matter what we do, I strongly advise against just adding a setting to take xDrip back to the way it was.  We changed the way xDrip was because it was a problem.

**Comment by lz2jvc** [comment]
Thanks for answering the second question. But for the first major problem from today's update for the two lines of text, I hope things will be fixed in the next update. It's not aesthetically pleasing if you leave things like that

**Comment by trallen** [comment]
To add to @Navid200 's excellent explanation, the new behaviour in the second item (that is, not rescaling the graph when predictions are added, but rather keeping the same view and adding the predictions off-screen to the right) can be disabled by turning off "Locked Time Period Always Used" (accessible by long-pressing the xDrip+ name at the top of the main screen. Of course, this may affect your view in other ways as well.

**Comment by Navid200** [comment]
As it turns out the cause of the change in the way sensor expiry is shown is not something any of us intended!

We have changed xDrip compatibility so that it can be installed on Android 15.  I tested it and never noticed this change.  This is caused by the target Android having changed from Android 6 to Android 7.

Thanks for reporting this.
I will be working on this to figure out how to fix it and it will be fixed soon.

**Reply by lz2jvc** [reply]
Thanks. I am the administrator of 2 large Bulgarian Facebook groups for diabetics. For this reason, I can provide one more photo of another Bulgarian xDrip+ user
![2024-05-30 10 58 58](https://github.com/NightscoutFoundation/xDrip/assets/168529053/5bdee63d-f706-48b8-9426-4074008750e9)

**Reply by Navid200** [reply]
This will be fixed soon.
